tests/src/python/test_qgsserver_wms.py

```diff
@@ -1865,6 +1865,19 @@ def test_wms_GetLegendGraphic_BBox2(self):
         r, h = self._result(self._execute_request(qs))
         self._img_diff_error(r, h, "WMS_GetLegendGraphic_BBox2")
 
+    def test_wms_GetProjectSettings_wms_print_layers(self):
+        projectPath = self.testdata_path + "test_project_wms_printlayers.qgs"
+        qs = "?" + "&".join(["%s=%s" % i for i in list({
+            "MAP": projectPath,
+            "SERVICE": "WMS",
+            "VERSION": "1.3.0",
+            "REQUEST": "GetProjectSettings"
+        }.items())])
+        header, body = self._execute_request(qs)
+        xmlResult = body.decode('utf-8')
+        self.assertTrue(xmlResult.find("<WMSBackgroundLayer>1</WMSBackgroundLayer>") != -1)
+        self.assertTrue(xmlResult.find("<WMSDataSource>contextualWMSLegend=0&amp;crs=EPSG:21781&amp;dpiMode=7&amp;featureCount=10&amp;format=image/png&amp;layers=public_geo_gemeinden&amp;styles=&amp;url=https://qgiscloud.com/mhugent/qgis_unittest_wms/wms?</WMSDataSource>") != -1)
+        self.assertTrue(xmlResult.find("<WMSPrintLayer>contextualWMSLegend=0&amp;amp;crs=EPSG:21781&amp;amp;dpiMode=7&amp;amp;featureCount=10&amp;amp;format=image/png&amp;amp;layers=public_geo_gemeinden&amp;amp;styles=&amp;amp;url=https://qgiscloud.com/mhugent/qgis_unittest_wms_print/wms?</WMSPrintLayer>") != -1)
```

Comment:
(minor comment) If you rebase your branch, you'll see there's a new function for testing `GetProjectSettings` stuff:

```
def test_getprojectsettings(self):
    self.wms_request_compare('GetProjectSettings')
```

Maybe it would be more homogenous to test this new feature with the same mechanism. Moreover, it'd allow to test the whole response. What do you think?

Reply:
I think in this case it is better to only test the new elements. Because in the project, there is a WMS layer pointing to QGIS cloud. If testing the whole response, any change in the server behaviour (or if the server is down) will make the test fail.
I'm considering using the new function 'test_getprojectsettings' for future tests. Is it possible to use it with any project file? It seems self.wms_request_compare always uses test_project.qgs

Reply:
>  Is it possible to use it with any project file? It seems self.wms_request_compare always uses test_project.qgs

Yes, `wms_request_compare` should be tweaked to do that.